### DISCUSSION TITLE
SmartAutoClicker-2.4.0

### DISCUSSION THREAD

**Discussion by Nain57** [Nain57/Smart-AutoClicker#349] [Announcements]
# New Features

## Simple scenarios 
When creating a new scenario, you can now select between 2 types, simple and smart. While a smart scenario is the regular Smart AutoClicker experience, the simple scenario is more like the regular AutoClicker experience, without image detection. This will allow you to create a scenario when willing to automate simple use cases.

## Ui improvements
* New actions overlay look and feel for simple scenario is also applied to smart ones.
* Indications during click/swipe selections are now hidding automatically after 3s to let you see the top of the screen.
* When updating a click/swipe position, previous selection is now visible.
* All floating menu now shares the same position, no need to move them one by one anymore.
* Add the possibility the stop a scenario by pressing "Volume Down"

# Bug fixes
* Fix a crash on Android 12 & 13 to due invalid notification icon. This has been introduced with the previous version with the same fix but for Android 9 devices. We now use old format for old phones and new format for new ones.
* Fix a crash when the user start a scenario containing a condition bigger than the selected detection area. User should no longer be able to achieve that by moving the selector really quickly.
* Fix a possible crash during a rotation with the debug enabled.
* Fix a possible Application Not Responding when closing all overlays at once.
* Catch and display an error message when smart autoclicker native lib is not found on the device. This should not happen when installing via regular means, but if anybody encounter this, it will tell them why.

<hr /><em>This discussion was created from the release <a href='https://github.com/Nain57/Smart-AutoClicker/releases/tag/2.4.0'>SmartAutoClicker-2.4.0</a>.</em>

**Comment by muhdf4iz** [comment]
Thanks for your great app. It is awesome. Actually I am waiting for a feature to run a scenario with specific time. Right now I am using floating clock to capture a specific time to run. Is there any plan to implement this feature? thanks
